### PR TITLE
Notify cashbox removal

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Notify operator on cashbox removed](https://trello.com/c/fqyEXyFB/765-notify-operator-on-cashbox-removed)

File: migrations/1621430588944-notify-cashbox-removal.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      if(config.notifications_notificationCenter_active) {
+        newConfig.notifications_notificationCenter_security = true
+      }
+
+      return saveConfig(newConfig)
+        .then(() => db.multi(sql, next))
+        .catch(err => {
+          if (err.message === 'lamassu-server is not configured') {
+            return next()
+          }
```
Comment (josepfo) [inline migrations/1621430588944-notify-cashbox-removal.js]: This condition only makes sense when migrating something from v7.4. I did the same mistake hehe
```suggestion
```

**Review by josepfo — CHANGES_REQUESTED** [review]

File: migrations/1621430588944-notify-cashbox-removal.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      if(config.notifications_email_active) {
+        newConfig.notifications_email_security = true
+      }
+      if(config.notifications_sms_active) {
+        newConfig.notifications_sms_security = true
+      }
+      if(config.notifications_notificationCenter_active) {
+        newConfig.notifications_notificationCenter_security = true
+      }
+
```
Comment (josepfo) [inline migrations/1621430588944-notify-cashbox-removal.js:21]: Don't we want to add these fields when the value is false as well?

**Review by RafaelTaranto — APPROVED** [review]

File: lib/notifier/index.js [hunk 3]
```diff
@@ -246,6 +246,41 @@ function sendTransactionMessage (rec, isHighValueTx) {
...
+
+      const emailActive =
+        notifications.email.active &&
+        notifications.email.security
+
+      const smsActive =
+        notifications.sms.active &&
+        notifications.sms.security
+      
+      if (emailActive) promises.push(emailFuncs.sendMessage(settings, rec))
```
Comment (joshmh) [inline lib/notifier/index.js:276]: This notification logic should be general purpose for all notifications. @chaotixkilla can you create a new Trello card to generalize the logic?

**Pull request merged**